## Task 14 — Customer Traffic by Subscription Plan

### Business Question

Which subscription plans have the highest average download and upload traffic per customer?

### Analysis

Group customers by subscription plan, calculate the number of customers and average download and upload traffic for each plan, then calculate average total traffic and sort the plans in descending order.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
customer_traffic = starlink.groupby("plan_name").agg(
    users_amount = ("unique_id","count"),
    gb_downloaded_average = ("gb_downloaded","mean"),
    gb_uploaded_average = ("gb_uploaded","mean")
)
customer_traffic["average_total_traffic"] = customer_traffic["gb_downloaded_average"] + customer_traffic["gb_uploaded_average"]
customer_traffic_sorted = customer_traffic.sort_values(
    "average_total_traffic",
    ascending = False
)
print(customer_traffic_sorted)

             users_amount  gb_downloaded_average  gb_uploaded_average  \
plan_name                                                               
Business             1019            1546.626850           268.643808   
Roam                 2078            1531.217988           263.877743   
Residential          5959            1507.878364           257.777402   
Mini                  944            1503.507140           258.366981   

             average_total_traffic  
plan_name                           
Business               1815.270658  
Roam                   1795.095731  
Residential            1765.655766  
Mini                   1761.874121  


### Result

The Business plan has the highest average total traffic at 1,815.27 GB per customer, followed by Roam at 1,795.10 GB and Residential at 1,765.66 GB. Mini has the lowest average total traffic at 1,761.87 GB per customer.

### Business Insight

Business customers demonstrate the highest average network activity, with both download and upload traffic exceeding the other subscription plans. This may indicate higher data usage among Business customers and could be relevant when evaluating plan pricing and customer behavior.